Boon command and gods data refactor

sneakyteak opened the pull request:
- Fixes bug in the boon command that caused `ecstaticObsession` and `ecstatic obsession reqs` to work but `ecstaticObsession reqs` to fail
- De-dupes fragile code that made this possible and cleans up boon command overall
- Moves all bot-output-string-formatting logic to /commands/utils and removes all references to formatters in /data
- Removes obsolete `other` property from the god type definition and gods' data
- Defines a utility function for gods to populate the `elements` property, makes use of it to populate `elements` for all gods, and makes this property mandatory henceforth
- Adds a few scenarios to test fixed / improved boons command

sneakyteak committed:
Fix syntax bug in boon requirements command

Previously, the boon command supported both user requests for boon info
in the form '!boon name' and '!boonName' with any capitalization.
However, the deeper boon requirements command was failing to recognize
'!boonName reqs', responding instead only to '!boon name reqs' with
spacing.

The bug, it turns out, is that we construct the boon name matcher twice;
once for boon info in the abilityExpressionMap, and once again for boon
reqs info in abilityPrereqMap. In the ExpressionMap, we converted
whitespace to ' *' 0-or-more matcher, but failed to do so in the
PrereqMap.

This commit fixes this issue. However, this betrays an unnecessary
fragility of the code. It would be better to construct these matching
strings one time and then to share them across both structures.

sneakyteak committed:
De-dupes boon command matcher logic

Fixes the pre-existing inconsistency between '!boon' and '!boon reqs'
commands in perpetuity by de-duping the matcher logic. This did require
a tiny touch to both the abilityFormatter utility and the god command
logic.

Next, we can clean up all the outdated formatted ability logic in the
gods' data and, likely, move the formatters from data to commands/util.

sneakyteak committed:
Removes all formatter use in data/

Removes all references to string formatter logic in data/ and moves
formatters to commands/utils.

sneakyteak committed:
Remove obsolete god param: other

We no longer list "other"-type boons for a god based on this param,
instead constructing it in the commands based on type and value data (to
exclude Duo boons).

sneakyteak committed:
Populate elements for all gods

Following the example in Aphrodite's data, populate elements for all
gods. Rather than duping the logic in each god, hoists it as a utility
function exported from god.ts. As all gods now have an elements
property, the property was made mandatory.

Ellisande reviewed (approved):
Looks good, my only suggestion would be to add a test or two for some regular boons to make sure their formatting works correctly since your changes potentially affect them

sneakyteak committed:
Adds several scenarios for god boon requests

As this refactor touched commands/god in addition to the changes in
boonCommand, it is possible the update could change or break behavior
for godCommand. This adds several tests that verify asking for boon info
through the godCommand handler still works appropriately.
